src/scripts/domain/moveCalculator.ts

```
@@ -126,4 +126,199 @@ export class MoveCalculator {
+  /**
+   * Calculates all the possible attack moves for a given piece
+   * @returns Move array if there are any available moves, otherwise null
+   */
+  public static calculateAttackMoves(
+    board: BoardInfo,
+    piece: PieceInfo,
+  ): Move[] | null {
+    const moves: Move[] = [];
+
+    if (piece.isPromoted) {
+      this.addKingAttackMoves(board, moves, piece);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)